fsw/cfe-core/src/es/cfe_es_resource.c

```diff
@@ -0,0 +1,428 @@
+/*
+**  GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**  Copyright (c) 2006-2019 United States Government as represented by
+**  the Administrator of the National Aeronautics and Space Administration.
+**  All Rights Reserved.
+**
+**  Licensed under the Apache License, Version 2.0 (the "License");
+**  you may not use this file except in compliance with the License.
+**  You may obtain a copy of the License at
+**
+**    http://www.apache.org/licenses/LICENSE-2.0
+**
+**  Unless required by applicable law or agreed to in writing, software
+**  distributed under the License is distributed on an "AS IS" BASIS,
+**  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**  See the License for the specific language governing permissions and
+**  limitations under the License.
+*/
+
+/*
+**  File:
+**  cfe_es_resource.c
+**
+**  Purpose:
+**      Function definitions related to CFE resource management
+**
+**  References:
+**     Flight Software Branch C Coding Standard Version 1.0a
+**     cFE Flight Software Application Developers Guide
+*/
+
+/*
+** Includes
+*/
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "cfe_platform_cfg.h"
+#include "cfe_es_resource.h"
+
+
+/*********************************************************************/
+/*
+ * CFE_ES_ResourceID_ToIndex_Internal
+ *
+ * For complete API information, see prototype in header
+ */
+int32 CFE_ES_ResourceID_ToIndex_Internal(uint32 Serial, uint32 TableSize, uint32 *Idx)
+{
+    if (Serial > CFE_ES_RESOURCEID_MAX)
+    {
+        return CFE_ES_ERR_RESOURCEID_NOT_VALID;
+    }
+
+    *Idx = Serial % TableSize;
```

Comment:
The 'Serial' variable is checked, should we also check for a non NULL pointer? I assume the check is not needed because the function is internal.

Reply:
Hmm, that may be a valid point ... In this case this function is internal-only but it is performing the range check on behalf of the various type-specific (public) API calls - i.e. so the same test doesn't need to be repeated 5 times over.

But you are correct that the parent doesn't validate the Idx pointer either.  I can add this.

Reply:
Update to check Idx in commit 4356693